simple-web-server/src/main.rs

```diff
@@ -29,4 +35,64 @@ fn check_port_availability(port: u16) -> bool {
         Ok(_) => true,
         Err(_) => false
     };
+}
+
+fn handle_connection(mut stream: TcpStream) {
+    let mut buffer = [0; 512];
+
+    stream.read(&mut buffer).unwrap();
+
+    // Prints a string of the HTTP request to Terminal
+    let http_str = String::from_utf8_lossy(&buffer[..]);
+    println!("{}", http_str); // prints out http request TODO: remove this later
+
+    let token = tokenize_http_req(http_str);
+    println!("{}", token.method); // TODO: Remove this later
+
+
+    let response = "HTTP/1.1 200 OK"; // Generic success response (placeholder)
+
+    stream.write(response.as_bytes()).unwrap();
+    stream.flush().unwrap();
+}
+
+// Basic version
+// TODO: maybe add user_agent, content_length, connection in the future
+struct HttpReq {
+    method: String,
+    // host: String,
+    // content_type: String,
+    // body: String,
+}
+
+// Early version, currently only looks for GET,POST,PUT,DELETE
+fn tokenize_http_req(req: Cow<'_, str>) -> HttpReq {
+
+    // splits string into newlines so it can be iterated over
+    let header = req.lines();
+    let mut method = String::new();
+
+    for line in header {
+        let words = line.split_whitespace();
```

Comment:
Isn't the request method always included on the first line of the request? Can imagine a loop here might get quite expensive if the request is stuffed to the brim with headers

Reply:
Yeh it'll get expensive. I don't know exactly what we need tbh this was just something I thought we might need. There's a bunch of actual http parsers on the package manager. But Grant said he wanted to avoid using too many packages

Reply:
I think we should pull out the parsing into a separate parsing.rs file. We'll want to setup parsing for URIs, headers, json, etc.